### PR TITLE
[BugFix] fix create failed with CTAS

### LINKED ISSUE #19742: Unexpected behavior with CTAS
### Steps to reproduce the behavior (Required)

```sql
create table if not exists tbl_test distributed by hash(c1) as select 1 as c1, 2 as c2;
select * from tbl_test;

create table if not exists tbl_test distributed by hash(c1) as select 1 as c1, 2 as c2;
select * from tbl_test;

create table if not exists tbl_test distributed by hash(c1) as select 1 as c1, 2 as c2;
select * from tbl_test;

```

### Expected behavior (Required)
Always return same result from select statement
```sql
+----+----+
| c1 | c2 |
+----+----+
| 1  | 2  |
+----+----+
```

### Real behavior (Required)
```sql
+----+----+
| c1 | c2 |
+----+----+
| 1  | 2  |
| 1  | 2  |
| 1  | 2  |
+----+----+
```

### StarRocks version (Required)
- 2.5

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19742

## Problem Summary(Required) ：

If creating table failed with any reason, should not execute the data population statement .

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 7 (57.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[167]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|2|50.00%|[776]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|2|50.00%|[592]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateTableAsSelectStmt.java|1|1|100.00%|[]|

**Comment by dirtysalt** [comment]
The reason why I make it as enum instead of constant is, if it's defined as constant, the header file and cpp file are following, they are not constant values at all.

```
namespace starrocks {

class InternalServiceConstants {
 public:
  InternalServiceConstants();

  int32_t INVALID_PLAN_NODE_ID;
  int32_t FUNC_VERSION_UNIX_TIMESTAMP_INT64;
};

extern const InternalServiceConstants g_InternalService_constants;

} // namespace

---

#include "InternalService_constants.h"

namespace starrocks {

const InternalServiceConstants g_InternalService_constants;

InternalServiceConstants::InternalServiceConstants() {
  INVALID_PLAN_NODE_ID = -1;

  FUNC_VERSION_UNIX_TIMESTAMP_INT64 = 5;

}

} // namespace

```


But if I define it as enum, it's constant

```
struct TFunctionVersion {
  enum type {
    FUNC_VERSION_UNIX_TIMESTAMP_INT64 = 5
  };
};

```

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 7 (57.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[167]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|2|50.00%|[776]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|2|50.00%|[592]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateTableAsSelectStmt.java|1|1|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by dirtysalt** [comment]
run starrocks_admit_test

**Comment by dirtysalt** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java [hunk 1]
```diff
@@ -163,7 +163,8 @@ default void dropTable(DropTableStmt stmt) throws DdlException {
     default void alterTable(AlterTableStmt stmt) throws UserException {
     }
 
-    default void createTable(CreateTableStmt stmt) throws DdlException {
+    default boolean createTable(CreateTableStmt stmt) throws DdlException {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:166]: Please comment when to return true or not.

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java (hunk 1 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:166]: I write a comment in the implementation instead of here

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**